src/Generator.php

```diff
@@ -10,10 +10,10 @@
 {
     /**
      * @param int The generation size
-     * @param callable  a rand() function
+     * @param Random\RandomRange
      * @return GeneratedValueSingle<T>
      */
-    public function __invoke($size, $rand);
+    public function __invoke($size, Random\RandomRange $rand);
```

Comment:
It would be more stable for them to depend on an interface but it can be extracted later with the same name moving `RandomRange` to another class name. Wouldn't do it here as the change is already large.